### PR TITLE
feat: Introduce pluggable inter-flow dispatch policy framework

### PR DESCRIPTION
**Pull request by LukeAVanDrie**
This work #674 

This pull request introduces the `InterFlowDispatchPolicy` framework, the third and final major component of the new pluggable flow control system outlined in our design proposals. This framework is responsible for **inter-flow fairness**, deciding *which flow's queue* gets the next opportunity to dispatch a request to the scheduler.

This change completes the two-tier policy engine by adding the strategic scheduling layer that complements the existing `IntraFlowDispatchPolicy` (which handles tactical, intra-queue scheduling). This allows operators to configure different fairness and utilization strategies based on their needs.

### What this PR does

-   **Introduces `framework.InterFlowDispatchPolicy`**: The core contract that all inter-flow fairness plugins must implement. Its primary method, `SelectQueue`, inspects a `PriorityBandAccessor` and returns the chosen queue.

-   **Introduces `framework.PriorityBandAccessor`**: A safe, read-only abstraction that provides policies with a view into the state of all flow queues within a single priority level, without exposing mutable state.

-   **Provides a Plugin Factory**: A standard registration and instantiation mechanism (`dispatch.MustRegisterPolicy`, `dispatch.NewPolicyFromName`) allows new policies to be discovered and loaded by name, consistent with the existing `SafeQueue` and `IntraFlowDispatchPolicy` frameworks.

-   **Adds a Conformance Test Suite**: A new functional test (`TestInterFlowDispatchPolicyConformance`) automatically validates the basic interface contract for all registered policy plugins, ensuring they handle edge cases like nil or empty priority bands correctly.

-   **Implements `roundrobin` Policy**: A stateful, fair-queuing policy that cycles through non-empty queues sequentially. It is resilient to dynamic changes in the set of available flows and serves as a robust default for ensuring fairness between tenants or models.

-   **Implements `besthead` Policy**: A stateless, greedy policy that selects the queue containing the single highest-priority item (based on each queue's `ItemComparator`). This policy is useful for maximizing resource utilization when strict fairness is not the primary concern. It also validates that it is only comparing items with compatible priority types.

### CODE REVIEW

File: pkg/epp/flowcontrol/framework/plugins/policies/intraflow/dispatch/README.md [hunk 1]
```diff
@@ -10,9 +10,9 @@ The `controller.FlowController` uses a two-tier policy system to manage requests
 plugins represent the first tier, making tactical decisions about the ordering of requests *within* a single logical
 flow (e.g., for a specific model or tenant).
 
-This contrasts with the `framework.InterFlowDispatchPolicy` (not yet implemented), which is responsible for
-**spatial fairness**: deciding *which flow's queue* gets the next opportunity to dispatch a request. The
-`framework.IntraFlowDispatchPolicy` only operates *after* the inter-flow policy has selected a specific queue.
+This contrasts with the `framework.InterFlowDispatchPolicy`, which is responsible for deciding *which flow's queue*
```
Comment (Xunzhuo) [inline pkg/epp/flowcontrol/framework/plugins/policies/intraflow/dispatch/README.md:13]: Is this file duplicated?

Clipboard_Screenshot_1752749548

Comment (kfswain) [reply pkg/epp/flowcontrol/framework/plugins/policies/intraflow/dispatch/README.md:13]: Luke makes a distinction between `inter-` & `intra-` flow. So those readme's would be distinct.

Comment (LukeAVanDrie) [reply pkg/epp/flowcontrol/framework/plugins/policies/intraflow/dispatch/README.md:13]: Inter and intra flow plugins have fairly similar contracts (SelectQueue vs SelectItem) and are organized similarly, but I wanted a distinct contributor guide colocated near the actual implementations for each extension point.

I could merge these all into one doc under framework or framework/plugins, but since contributors do not need to understand every extension point to implement one (e.g., adding a new inter flow fairness policy does not require understanding the intra flow policies), I thought the current approach was less overwhelming.

**Comment by kfswain** [comment]
/lgtm
/approve

Exciting to see this come together!
